azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/inbox.py

```diff
@@ -0,0 +1,27 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module contains an Inbox class for use with a synchronous client."""
+
+from six.moves import queue
+
+
+class Inbox(queue.Queue):
+    """Queue-like object, used to recieve Messages.
+
+    Has two Queue interfaces, for use with different parts of the application.
+    Do Queue operations on the interfaces rather than the object itself.
+
+    :ivar manager_interface: The interface for synchronous, threadsafe operations
+    within the InboxManager. Same API as queue.Queue.
+    :ivar client_interface: The interface for synchronous, threadsafe operations
+    within the Client. Same API as queue.Queue.
+    """
+
+    def __init__(self):
+        """Initializer for Inbox."""
+        super().__init__()
+        self.manager_interface = self
```

Comment:
>manager_interface [](start = 13, length = 17)

Should these attributes start with an underscore?

Reply:
No, they get accessed by other classes (the manager and client respectively)

---
In reply to: [262233694](https://github.com/Azure/azure-iot-sdk-python-preview/pull/55#discussion_r262233694) [](ancestors = 262233694)